### DISCUSSION TITLE
Support of non-UTF-8 path

### DISCUSSION THREAD

**Discussion by vincent-herlemont** [spacedriveapp/spacedrive#666] [General]
It seems that the code in [`file_path_helper.rs` L69](https://github.com/spacedriveapp/spacedrive/blob/8d20fcfe33bfb16f9eb1e1e12de3170661b905bb/core/src/location/file_path_helper.rs#L69-L69) removes support for non-UTF-8 paths in the Space Drive application. I have some questions about this:

- Is this not an issue for users with file paths containing non-UTF-8 characters?
- If it is an issue, do you have plans to support non-UTF-8 paths in the future?
- Have you considered using a library like [camino](https://docs.rs/camino/latest/camino) to avoid using [`.expect("Found non-UTF-8 path")`](https://github.com/spacedriveapp/spacedrive/search?q=.expect%28%22Found+non-UTF-8+path%22%29)?

**Comment by brxken128** [comment]
It's not just the code you mention that "removes" support for non-UTF8 paths - similar patterns are used all throughout the codebase, as non-UTF8 paths are *extremely* uncommon (and because of this, we likely don't need to support them). I'd argue that Rust doesn't have great compatibility with non-UTF8 paths either.

As far as I can tell, `camino` doesn't aid supporting non-UTF8 paths, but in fact strictly prohibits them from even existing - your question is a little conflicting in that regard.

On another note, I don't think `camino` would be much help here - path conversions aren't the nicest, but porting our code to the `Utf8Path*` types probably won't be much fun either. I'd be willing to give it a shot if other members of the team deem it beneficial though!
